### DISCUSSION TITLE
Retrieve indexes only

### DISCUSSION THREAD

**Discussion by brandon-wesley** [dexie/Dexie.js#1269] [Q&A] [answered]
Hello,

I'm wondering, is there a way to retrieve only the primary keys and indexed properties of a collection, instead of loading the entire records (for performance reasons)?

I see the `Collection.keys()` option, but this seems to only retrieve one key at a time, and I am looking to get all of them.

**Comment by dfahlander** [comment, marked as answer]
There is no such possibility in the IndexedDB API. The closest you could come is to use a compound index that contains the properties you need and retrieve all those indexes.

```js
const db = new Dexie('dbname');
db.version(1).stores({
  myTable: 'id, [id+firstName+lastName+age]'
});

const result = await db.myTable.orderBy('[id+firstName+lastName+age]').keys();

```

The `result` would give you:

```js
[["id1", "firstName1", "lastName1", 21], ["id2", ...]]
```

**Reply by brandon-wesley** [reply]
Too bad...

Thanks for the awesome library 👍🏼

**Reply by dfahlander** [reply]
I don't know what performance reasons you are exactly after, but i suppose you need to store heavy data in the object.

If so, Blobs can also be a burden lifter in that sense as their contents are not loaded until you start reading from them.

**Reply by brandon-wesley** [reply]
I'll look into that - thanks!
